src/server/script_mgr.cc

```diff
@@ -192,28 +194,47 @@ io::Result<optional<ScriptMgr::ScriptParams>, GenericError> DeduceParams(string_
   return params;
 }
 
+unique_ptr<char[]> CharBufFromSV(string_view sv) {
+  auto ptr = make_unique<char[]>(sv.size() + 1);
+  memcpy(ptr.get(), sv.data(), sv.size());
+  ptr[sv.size()] = '\0';
+  return ptr;
+}
+
 io::Result<string, GenericError> ScriptMgr::Insert(string_view body, Interpreter* interpreter) {
   // Calculate hash before removing shebang (#!lua).
   char sha_buf[64];
   Interpreter::FuncSha1(body, sha_buf);
   string_view sha{sha_buf, std::strlen(sha_buf)};
 
-  auto params = DeduceParams(&body);
-  if (!params)
-    return params.get_unexpected();
+  string_view orig_body = body;
+
+  auto params_opt = DeduceParams(&body);
+  if (!params_opt)
+    return params_opt.get_unexpected();
+  auto params = params_opt->value_or(default_params_);
+
+  // If the script is atomic, check for possible squashing optimizations.
+  // For non atomic modes, squashing increases the time locks are held, which
+  // can decrease throughput with frequently accessed keys.
+  optional<string> async_body;
+  if (params.atomic) {
+    if (async_body = Interpreter::DetectPossibleAsyncCalls(body); async_body)
```

Comment:
following what we discussed, please put it under flag. I am fine with flipping the flag to on but lets have some guard.